rdflib/parser.py

```diff
@@ -47,9 +47,43 @@
     "URLInputSource",
     "FileInputSource",
     "PythonInputSource",
+    "iri2uri",
 ]
 
 
+def iri2uri(iri: str) -> str:
+    """
+    Convert an IRI to a URI (Python 3).
+    https://stackoverflow.com/a/42309027
+    https://stackoverflow.com/a/40654295
+    netloc should be encoded using IDNA;
+    non-ascii URL path should be encoded to UTF-8 and then percent-escaped;
+    non-ascii query parameters should be encoded to the encoding of a page
+    URL was extracted from (or to the encoding server uses), then
+    percent-escaped.
+    >>> iri2uri("https://dbpedia.org/resource/Almería")
+    'https://dbpedia.org/resource/Almer%C3%ADa'
+    """
+
+    (scheme, netloc, path, query, fragment) = urlsplit(iri)
+
+    # Just support http/https, otherwise return the iri unmolested
+    if scheme not in ["http", "https"]:
+        return iri
+
+    scheme = quote(scheme)
+    netloc = netloc.encode("idna").decode("utf-8")
+    path = quote(path)
+    query = unquote(quote(query))
```

Comment:
I'm busy adding some more tests, but as far as I can tell this should not unquote after quoting, as then the following tests fails:

```console
_________________________________________________________ test_iri2uri[http://example.com/?\xe9=1-expected_result5] _________________________________________________________
Traceback (most recent call last):
  File "/home/iwana/sw/d/github.com/iafork/rdflib.cleanish/test/test_util.py", line 626, in test_iri2uri
    assert result in expected_result
AssertionError: assert 'http://example.com/?é=1' in {'http://example.com/?%C3%A9%3D1', 'http://example.com/?%C3%A9=1'}
```

I would say it should be:

```suggestion
    query = quote(query)
```

Reply:
I will make the change in a commit shortly, if you disagree let me know, I may be missing something.

Reply:
idk, looks like a typo,  so long ago, I've forgotten.
lgtm

Reply:
okay, resolving.